src/main/scala/com/microsoft/hyperspace/util/HyperspaceConf.scala

```diff
@@ -71,6 +71,20 @@ object HyperspaceConf {
       .toBoolean
   }
 
+  def fileBasedSourceBuilders(spark: SparkSession): String = {
+    spark.sessionState.conf
+      .getConfString(
+        "spark.hyperspace.index.sources.fileBasedBuilders",
+        "com.microsoft.hyperspace.index.sources.default.DefaultFileBasedSourceBuilder")
```

Comment:
Why this config is necessary? Can we just create all builders?

Reply:
To control what builders are being loaded similar to how plugins, listeners, etc. are loaded in `SparkContext`. What's the benefit of creating all builders available? The only benefit I can think of is not having to specify this config, but having a tight control seems better to me.